### PR TITLE
New base class for mGST and improvements to mGST+GHZ

### PR DESCRIPTION
**Pull request by rabrie**
Planned changes include:
- mGST operable under the new base class only
- Various adaptations and improvements to the mGST benchmark
- Splitting of mGST source file into `compressive_gst` and `gst_analysis` 
- Use of the BenchmarkObservationIdentifier class for the GHZ benchmark
- Addition of a plot for the GHZ benchmark
- Removal of DD code from utils.py (it was already commented out)
- Creation of a documented `example_ghz.ipynb` as a tutorial

### CODE REVIEW

File: CHANGELOG.rst
Comment (pedrofigro) [inline CHANGELOG.rst:1]: We should have a tag with this MR too, I think

File: examples/Outputs/IQMFakeApollo/20241206-100720/compressive_GST/IQMFakeApollo_20241206-100720_compressive_GST_configuration.json
Comment (pedrofigro) [inline examples/Outputs/IQMFakeApollo/20241206-100720/compressive_GST/IQMFakeApollo_20241206-100720_compressive_GST_configuration.json:1]: These output files keep sneaking into the commits somehow

File: examples/example_ghz.ipynb
Comment (pedrofigro) [inline examples/example_ghz.ipynb:1]: The preamble is nice; I'd reckon we should either make all other notebooks tutorial style or write readmes for each technique?

Comment (pedrofigro) [inline examples/example_ghz.ipynb:1]: Also perhaps it's best to clear all the outputs

File: examples/example_gst.ipynb
Comment (pedrofigro) [inline examples/example_gst.ipynb:1]: GST I reckon is good in this more tutorial style

File: examples/Outputs/IQMFakeApollo/20241206-100720/compressive_GST/IQMFakeApollo_20241206-100720_compressive_GST_configuration.json
Comment (freetonik) [reply examples/Outputs/IQMFakeApollo/20241206-100720/compressive_GST/IQMFakeApollo_20241206-100720_compressive_GST_configuration.json:1]: Add `examples/Outputs/*` to `.gitignore`

File: examples/example_ghz.ipynb
Comment (freetonik) [reply examples/example_ghz.ipynb:1]: Tutorial style is nice, IMO. We can also run notebooks as e2e tests, thus verifying both the correctness of the documentation and the correctness of the code.

Comment (freetonik) [reply examples/example_ghz.ipynb:1]: The outputs can be preserved for nicer rendered documentation. Notebooks can be viewed by users without then need to run them, which is nice. We could also render the jupyter notebooks within Sphinx documentation.

**Pull request merged**